### PR TITLE
Polish code in entry.go

### PR DESCRIPTION
**Pull request by wenovus**
- Add a few tests.
- Remove redunant code (`entryCache[n] = e`).
- Remove unreachable code (`if e.Kind == LeafEntry`)
- Improve naming and complexity a little.
- Improve comments

### CODE REVIEW

File: pkg/yang/entry.go
Comment (robshakir) [inline pkg/yang/entry.go:74]: Directory, leaf or leaf-list? I wonder whether it's best to say `schema tree node, which can be a directory (containing a subtree) or a leaf node (which contains YANG types that have no children, e.g., leaf, leaf-list)`.

File: pkg/yang/entry.go
Comment (robshakir) [inline pkg/yang/entry.go:1485]: Was this unused?

**Review by robshakir — APPROVED** [review]

File: pkg/yang/entry.go
Comment (wenovus) [reply pkg/yang/entry.go:1485]: This is used, I simplified it and took it out of the function.

File: pkg/yang/entry.go
Comment (wenovus) [reply pkg/yang/entry.go:74]: Done.

File: pkg/yang/entry.go
Comment (wenovus) [reply pkg/yang/entry.go]: `Errors`. I'm adding it to the comment.

**Comment by wenovus** [comment]
@robshakir for simplicity I put all the PRs in a chain, so can't merge until the previous ones have been approved. Starts at #160.

**Pull request merged**
